### ISSUE TITLE
Lab 02 - knitting issue

### ISSUE THREAD

**Issue by mbrentli** [Watts-College/cpp-529-spr-2023#21]
Hi - I'm running into problems when trying to knit the rmd to html.  It takes forever to process and then eventually I get the following error:

```
Error in writeLines(enc2utf8(text), con, ..., useBytes = TRUE) : 
  Error writing to connection
Calls: <Anonymous> -> <Anonymous> -> write_utf8 -> writeLines
Error in setwd(oldwd) : cannot change working directory
Calls: <Anonymous> -> setwd
Execution halted
```

I tried commenting out some of the chunks with VarSearch that were just to assist with determining variables, but it doesn't seem to make a difference in the processing.

Is anyone else having a similar issue?  Or are there any troubleshooting ideas?

**Comment by mbrentli** [comment]
Hi @AntJam-Howell  - I'm having the trouble after I've completed the lab/added code.  To be honest, I didn't knit immediately after download, so I'm not sure of the difference or processing speed before/after.  I did, however, copy-paste the code to a brand new rmd to see if that would speed up the process.  It is slightly faster, but still gets the error.  I will email you the rmd shortly

**Comment by mbrentli** [comment]
@AntJam-Howell  Additionally, I just updated RStudio (because I've been putting off that update for awhile now) and reloaded and tried to knit again - same issue.

**Comment by AntJam-Howell** [comment]
Hi @mbrentli, 
I've taken a look at your RMD file and part of the problem troubleshooting is that you've gone back and incorporated all the various parts of the lab questions into the lab instruction part, so there is 3 different datasets (county 2017, county 2012, census tract 2017) operating in tandem and it can get confusing.  Trying to sift through the code, nothing jumps out immediately for why the map is so distorted in the lines of code you have (I offer a more concise solution to render the map correctly below). 

In terms of speed, it could be computer specific or specific.  To speed-up a bit, it helps to combine the code together via piping %>% as shown in the lab 2 instructions rather than separate lines of code.  

To render the map correctly, use the following lines of code: 

```
vars.to.use <- c( HHIncome = "B19049_001",
                  HouseValue = "B25077_001" )

CenDF <- get_acs( geography="county", 
                  year=2017, 
                  survey="acs5",
                  variables=vars.to.use,
                  geometry=TRUE,
                  shift_geo=TRUE )

```

```
CenDF<-CenDF %>% 
  select( -moe ) %>%  
  spread( variable, estimate ) %>%
  mutate( HHInc_HousePrice_Ratio = ( HouseValue/HHIncome ) )

```

```
## Plot
ggplot( CenDF ) +
  geom_sf( aes( fill = HHInc_HousePrice_Ratio ), color=NA ) +
  coord_sf( datum=NA ) +
  labs(title = "House-price-to-income ratio by county",
       subtitle = "Alaska and Hawaii are shifted and not to scale",
       caption = "Source: ACS 5-year, 2017",
       fill = "house-price-\nto-income ratio") +
  scale_fill_viridis( direction=-1 )

```

Rather than trying to prepare for each question in advance and running datasets in tandem,  I would copy these three chunks (along with a few other lines of code to get the Question 1 answer) and paste them, for instance, into Question 2 verbatim.  Then change the year from 2017 to 2012, etc. 

Hope that helps.

**Comment by mbrentli** [comment]
Hi @AntJam-Howell Before your reply, I was still trying to troubleshoot on my own and realized that I had saved the rmd file in my personal P:// drive on the secure ASU network.  I moved it out of the network and stored it on the actual device, and while it is still slow, it actually knits the html now - so that seems to have been some of the problem.

It shouldn't be any other kind of computer specific issue, that I can think of (RStudio is updated, and all packages are updated, running on 13th Gen Intel(R) Core(TM) i7 1.8 GHz with 32GB RAM).

Which leads to part of the cause also being related your suggestions - I am not a programmer or coder by any means and this PEDAMS program is the beginning of my R journey, so my code is definitely less than ideal.  I find that I often break things down into as small of chunks as possible so that I can try to see exactly what each piece is doing, and I know that I save the intermediary output to objects far too often.  Without being familiar enough with what's going on, I am hesitant to combine or clean things up on my own (especially on such a short timeline when it takes 4 days to just get it to work the first time and only 2 days left until submission! haha) 

However, I will definitely look through your suggestions and try to figure out how to clean up what I can without breaking everything.  Thank you so much for your help!
